source/slang/lookup.cpp

```diff
@@ -410,9 +410,9 @@ void lookUpMemberImpl(
     if (auto declRefType = type->As<DeclRefType>())
     {
         auto declRef = declRefType->declRef;
-        if (auto assocTypeDeclRef = declRef.As<AssocTypeDecl>())
+        if (declRef.As<AssocTypeDecl>() || declRef.As<GlobalGenericParamDecl>())
```

Comment:
Hmm... Yeah, this seems to be a (small) argument for these two types sharing a base class, especially since after this line you go and case `declRef` to `ContainerDecl` which is the current shared based class between those two cases...